### PR TITLE
Add tests for WritableStreamDefaultController.signal

### PR DESCRIPTION
**Pull request by yutakahirano**
Tests for whatwg/streams#1132.

### CODE REVIEW

File: streams/writable-streams/aborting.any.js [hunk 1]
```diff
@@ -1376,3 +1376,99 @@ promise_test(t => {
...
+  assert_true(ctrl.signal.aborted);
+}, 'the abort signal is signalled synchronously - close');
+
+promise_test(async t => {
+  let ctrl;
+  const ws = new WritableStream({start(c) { ctrl = c; }});
+  const writer = ws.getWriter();
+
+  ctrl.error(TypeError());
+  await promise_rejects_js(t, TypeError, writer.closed);
```
Comment (MattiasBuelens) [inline streams/writable-streams/aborting.any.js:1432]: This should probably use `promise_rejects_exactly` and use a specific error object like `error1` or `error2`. That way, we won't get a false positive if an incorrect implementation rejects with a *different* `TypeError`.

(Same thing for the other tests.)

File: streams/writable-streams/aborting.any.js [hunk 2]
```diff
@@ -1376,3 +1376,99 @@ promise_test(t => {
...
+  const ws = new WritableStream({
+    start(c) { ctrl = c; },
+  });
+
+  ctrl.signal.addEventListener('abort', () => writer.abort(TypeError()));
+  const writer = ws.getWriter();
+  writer.abort(SyntaxError());
+  assert_true(ctrl.signal.aborted);
+
+  await promise_rejects_js(t, SyntaxError, writer.closed, 'closed');
```
Comment (MattiasBuelens) [inline streams/writable-streams/aborting.any.js:1473]: We should use `promise_rejects_exactly` here. Have one `abort()` call with `error1` and the other with `error2`, and check that we get `error1` at the end.

File: streams/writable-streams/aborting.any.js (hunk 1 above)
Comment (yutakahirano) [reply streams/writable-streams/aborting.any.js]: Done.

File: streams/writable-streams/aborting.any.js [hunk 3]
```diff
@@ -1376,3 +1376,104 @@ promise_test(t => {
...
+promise_test(async t => {
+  let ctrl;
+  const e1 = SyntaxError();
+  const e2 = TypeError();
+  const ws = new WritableStream({
+    start(c) { ctrl = c; },
+  });
+
+  ctrl.signal.addEventListener('abort', () => writer.abort(e2));
+  const writer = ws.getWriter();
```
Comment (ricea) [inline streams/writable-streams/aborting.any.js:1474]: As a C++ programmer, it feels weird to me when a variable is referenced before it is declared. Could you swap these two lines?

**Review by ricea — APPROVED** [review]
lgtm

File: streams/writable-streams/aborting.any.js (hunk 3 above)
Comment (yutakahirano) [reply streams/writable-streams/aborting.any.js:1478]: Done.

**Comment by yutakahirano** [comment]
Added tests for abortReason.

**Comment by domenic** [comment]
The following tests fail when I run this checkout against the reference implementation:

- `WritableStreamDefaultController.signal`
- `the abort signal is signalled synchronously - write`
- `the abort signal is signalled synchronously - close`
- `recursive abort() call`

**Review by domenic — APPROVED** [review]

**Pull request merged**